lib/model/message.dart

```diff
@@ -517,6 +578,40 @@ class MessageStoreImpl extends PerAccountStoreBase with MessageStore, _OutboxMes
   }
 }
 
+class _MarkReadOnScrollQueue {
+  _MarkReadOnScrollQueue();
+
+  bool get isEmpty => _queue.isEmpty;
+
+  final _set = <int>{};
+  final _queue = QueueList<int>();
+
+  /// Add [messageIds] to the end of the queue,
+  /// if they aren't already in the queue.
+  ///
+  /// Returns true if some were added (i.e. weren't already in the queue),
+  /// false otherwise.
+  bool add(Iterable<int> messageIds) {
```

Comment:
nit: containers like List and Set have methods `add` to add one element and `addAll` to add from an iterable, so probably clearest to match that naming:
```suggestion
  bool addAll(Iterable<int> messageIds) {
```